### PR TITLE
Fetch Alpha SMS delivery statuses

### PR DESCRIPTION
**Pull request by prabhanshuguptagit**
**Story card:** [sc-8985](https://app.shortcut.com/simpledotorg/story/8985/integrate-the-apis-of-new-bd-sms-provider)

## Because

Accompanying PR to https://github.com/simpledotorg/simple-server/pull/4272

## This addresses

Adds a daily job to fetch message statuses from alpha SMS.

### CODE REVIEW

**Comment by harimohanraj89** [comment]
Wanna base this off of the other branch for a cleaner diff?

**Comment by prabhanshuguptagit** [comment]
Woops, done!

File: app/jobs/alpha_sms_status_job.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  def perform(request_id)
+    detailable = AlphaSmsDeliveryDetail.find_by!(request_id: request_id)
+    response = Messaging::AlphaSms::Api.new.get_message_status_report(request_id)
+    raise_api_errors(request_id, response)
+    delivery_status_data = response.dig("data", "recipients").first["status"]
+
+    detailable.request_status = delivery_status_data if delivery_status_data
+    detailable.save!
+
+    # log(response)
```
Comment (PriyangaPKini) [inline app/jobs/alpha_sms_status_job.rb:21]: May be remove the comment?

**Review by harimohanraj89 — APPROVED** [review]
Looks solid. Is there some sandbox env that we can connect to, or will this work only in production?

**Pull request merged**